btc-lsp/src/BtcLsp/Thread/LnChanWatcher.hs

```diff
@@ -0,0 +1,25 @@
+module BtcLsp.Thread.LnChanWatcher
+  (watchChannelEvents, forkThread)
+where
+
+import BtcLsp.Import hiding (takeMVar, putMVar, newEmptyMVar)
+import UnliftIO.Concurrent (ThreadId, forkFinally, putMVar)
+import BtcLsp.Import (newEmptyMVar)
+import qualified LndClient.RPC.Silent as Lnd
+import BtcLsp.Storage.Model.LnChan (persistChannelUpdates)
+import LndClient
+
+forkThread :: MonadUnliftIO m => m () -> m (ThreadId, MVar ())
+forkThread proc = do
+    handle <- newEmptyMVar
+    tid <- forkFinally proc (\_ -> putMVar handle ())
+    return (tid, handle)
+
+watchChannelEvents :: (Storage m, KatipContext m) => LndEnv -> m (ThreadId, MVar ())
+watchChannelEvents lnd = forkThread $ withRunInIO act
+    where
+      act run = do
+        _ <- Lnd.subscribeChannelEvents (run . persistChannelUpdates) lnd
```

Comment:
There are missing parts:

- Re-subscribing in case of network/lnd failure
- Fetching channel data which changed while lsp subscription was offline for whatever reason